### PR TITLE
School Library: Testing

### PR DESCRIPTION
**Pull request by ryxtor**
## Project requirements
Add unit tests to your classes:

- `Book`
- `Person`
- `Student`
- `Teacher`
- `Classroom`
- `Rental`
- all classes that you created while implementing the Decorator design pattern

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+require './book'
+describe Book do
+  before :each do
+    @book = Book.new 'Title', 'Author'
+  end
```
Comment (julie-ify) [inline spec/book_spec.rb]: - [ ] Let ensure that there are unit tests for all the methods in all the classes, please kindly write unit test for the `add_rental` method in the Book class.

File: spec/classroom_spec.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+require './classroom'
+
+describe Classroom do
+  before :each do
+    @classroom = Classroom.new 'Label'
+  end
```
Comment (julie-ify) [inline spec/classroom_spec.rb]: - [ ] Let ensure that there are unit tests for all the methods in all the classes, please kindly write unit test for the `add_student` method in the Classroom class.

File: spec/nameable_spec.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+require './nameable'
+
+describe Nameable do
+  before :each do
+    @nameable = Nameable.new
+  end
```
Comment (julie-ify) [inline spec/nameable_spec.rb:6]: - [ ] Let ensure that there are unit tests for all the methods in all the classes, please kindly write unit test for the `correct_name` method in the Nameable class.

File: spec/person_spec.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+require './person'
+
+describe Person do
+  before :each do
+    @person = Person.new 20, 'Lucas'
+  end
```
Comment (julie-ify) [inline spec/person_spec.rb]: - [ ] Let ensure that there are unit tests for all the methods in all the classes, please kindly write unit test for the `can_use_services?`, `correct_name`, `add_rental`  methods in the Person class.

File: spec/student_spec.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
+require './student'
+
+describe Student do
+  before :each do
+    @student = Student.new 17, 'Francisco'
+  end
+  describe '#new' do
+    it 'Return a new Student object' do
+      expect(@student).to be_an_instance_of Student
+    end
```
Comment (julie-ify) [inline spec/student_spec.rb]: - [ ] Let ensure that there are unit tests for all the methods in all the classes, please kindly write unit test for the `add_classroom` method in the Student class.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @ryxtor and @franciscoPonceDev, 

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Tests passed ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Whoistolu — APPROVED** [review]
### Hi Team
@ryxtor and @franciscoPonceDev 

Great job implementing the recommended changes 👍🏽 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**